hooks/useRbfTransactionData/helpers.ts

```diff
@@ -0,0 +1,121 @@
+  const bigNumHigherFee = BigNumber(higherFeeRate);
+
+  return {
+    [lowerName]: {
+      enoughFunds: bigNumLowerFee.lte(BigNumber(stxAvailableBalance)),
+      feeRate: microstacksToStx(bigNumLowerFee).toNumber(),
+      fee: microstacksToStx(bigNumLowerFee).toNumber(),
+    },
+    [higherName]: {
+      enoughFunds: bigNumHigherFee.lte(BigNumber(stxAvailableBalance)),
+      feeRate: microstacksToStx(bigNumHigherFee).toNumber(),
+      fee: microstacksToStx(bigNumHigherFee).toNumber(),
+    },
+  };
+};
+
+export const sortFees = (fees: RbfRecommendedFees) =>
+  Object.fromEntries(
+    Object.entries(fees).sort((a, b) => {
+      const priorityOrder = ['highest', 'higher', 'high', 'medium'];
+      return priorityOrder.indexOf(a[0]) - priorityOrder.indexOf(b[0]);
+    }),
+  );
+
+export const calculateStxData = async (
+  transaction: StxTransactionData,
+  btcNetwork: SettingsNetwork,
+  stacksNetwork: StacksNetwork,
+  appInfo: AppInfo | null,
+  stxAvailableBalance: string,
+): Promise<RbfData> => {
+  const { fee } = transaction;
+  const txRaw: string = await getRawTransaction(transaction.txid, btcNetwork);
+  const unsignedTx: StacksTransaction = deserializeTransaction(txRaw);
+
+  const [slow, medium, high] = await estimateTransaction(unsignedTx.payload, undefined, stacksNetwork);
+
+  let feePresets: RbfRecommendedFees = {};
+  let mediumFee = medium.fee;
+  let highFee = high.fee;
+  const higherFee = fee.multipliedBy(1.25).toNumber();
+  const highestFee = fee.multipliedBy(1.5).toNumber();
+
+  if (appInfo?.thresholdHighStacksFee) {
+    if (high.fee > appInfo.thresholdHighStacksFee) {
+      // adding a fee cap
+      highFee = appInfo.thresholdHighStacksFee * 1.5;
+      mediumFee = appInfo.thresholdHighStacksFee;
+    }
+  }
+
+  let minimumFee = fee.multipliedBy(1.25).toNumber();
+  if (!Number.isSafeInteger(minimumFee)) {
+    // round up the fee to the nearest integer
+    minimumFee = Math.ceil(minimumFee);
+  }
+
+  if (fee.lt(BigNumber(mediumFee))) {
+    feePresets = constructRecommendedFees('medium', mediumFee, 'high', highFee, stxAvailableBalance);
+  } else {
+    feePresets = constructRecommendedFees('higher', higherFee, 'highest', highestFee, stxAvailableBalance);
+  }
+
+  return {
+    rbfTransaction: undefined,
+    rbfTxSummary: {
+      currentFee: microstacksToStx(fee).toNumber(),
+      currentFeeRate: microstacksToStx(fee).toNumber(),
+      minimumRbfFee: microstacksToStx(BigNumber(minimumFee)).toNumber(),
+      minimumRbfFeeRate: microstacksToStx(BigNumber(minimumFee)).toNumber(),
+    },
+    rbfRecommendedFees: sortFees(feePresets),
+    mempoolFees: {
+      fastestFee: microstacksToStx(BigNumber(high.fee)).toNumber(),
+      halfHourFee: microstacksToStx(BigNumber(medium.fee)).toNumber(),
+      hourFee: microstacksToStx(BigNumber(slow.fee)).toNumber(),
+      economyFee: microstacksToStx(BigNumber(slow.fee)).toNumber(),
+      minimumFee: microstacksToStx(BigNumber(slow.fee)).toNumber(),
```

Comment:
should we return micro stacks or stx values here? dunno if this util can be used in something else and then you need to do the map to micro stacks again.

Reply:
not sure if this util is going to be used in other circumstances, currently it feels like it's easier to convert it here instead of converting it in both mobile app and web extension in each place where it's used